supervision/detection/line_counter.py

```diff
@@ -113,13 +118,200 @@ def annotate(self, frame: np.ndarray, line_counter: LineZone) -> np.ndarray:
 
         Attributes:
             frame (np.ndarray): The image on which the line will be drawn.
-            line_counter (LineCounter): The line counter
-                that will be used to draw the line.
+            line_counter (LineZone): The line counter object used to annotate.
 
         Returns:
             np.ndarray: The image with the line drawn on it.
 
         """
+
+        def annotate_count(text, text_over=True):
```

Comment:
Let's not define that function inside another function. Let's convert this into a private `_annotate_count` method in the `LineZoneAnnotator` class.

Reply:
Let's also add argument types.